code/game/objects/effects/spawners/unique_spawner.dm

```diff
@@ -0,0 +1,129 @@
+/***
+ * How to use:
+ * Either place the base unique_spawner in the map and define the key_override to determine which spawners belong to one another.
+ * Or make a child instance of the unique_spawner and define the values there.
+ * When using the base version be sure to define both total_amount and max_per_spawner on only one of the spawners
+***/
+
+/obj/effect/spawner/unique_spawner
+	name = "unique spawner"
+	icon = 'icons/mob/screen_gen.dmi'
+	icon_state = "x2"
+	/// Which path will be used to spawn. Will be used to determine the uniqueness if you use the base unique_spawner type
+	var/path_to_spawn
+	/// How many will spawn. Set this in the map edit tool or in a child instance to set it for all the other instances.
+	var/total_amount
+	/// How many will spawn maximum per spawner. Set this in the map edit tool or in a child instance to set it for all the other instances.
+	var/max_per_spawner
+	/// The key of the group spawners this spawner belongs to. Define this for map instances if you want to link different spawners up or want to use the base version.
+	var/key_override
+
+	/// The assoc list of unique spawners. Key = type or key_override, value = list of spawners of that type
+	var/static/list/spawners
+	/// Which spawners are chosen to spawn the objects
+	var/static/list/chosen_spawners
+	/// The list which holds the total_amount values for each spawner group
+	var/static/list/total_amount_for_spawners
+	/// The list which holds the max_per_spawner values for each spawner group
+	var/static/list/max_per_spawner_for_spawners
+
+/obj/effect/spawner/unique_spawner/Initialize(mapload)
```

Comment:
**THANK YOU** 

If I see another spawner with `/New()` I am gonna be annoyed

Reply:
Haha. I was tempted to use it but I found other ways to circumvent using it.